plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -41,22 +41,25 @@
       "com.google.cloud.tools.eclipse.sdk/enable.managed.cloud.sdk";
 
   // To be able to write tests for the managed Cloud SDK feature, which is disabled at the moment.
+  // This will go away once we permanently enable the feature.
   @VisibleForTesting public static boolean forceManagedSdkFeature;
 
   @VisibleForTesting public static CloudSdkManager instance;
 
+  // readers = using SDK, writers = modifying SDK
+  private final ReadWriteLock modifyLock;
+
   public static synchronized CloudSdkManager getInstance() {
     if (instance == null) {
-      instance = new CloudSdkManager();
+      instance = new CloudSdkManager(new ReentrantReadWriteLock());
     }
     return instance;
   }
 
   @VisibleForTesting
-  CloudSdkManager() {}
-
-  // readers = using SDK, writers = modifying SDK
-  @VisibleForTesting final ReadWriteLock modifyLock = new ReentrantReadWriteLock();
+  CloudSdkManager(ReadWriteLock modifyLock) {
```

Comment:
I wonder how this will work with mocking.  When mocking a normal class, it seems everything is pass-through except for the explicitly mocked methods.  So it's possible that we'll hit NPEs.  Maybe not a bad thing.

Reply:
I think you are talking about [partial mocking](https://stackoverflow.com/questions/14970516/use-mockito-to-mock-some-methods-but-not-others), and looks like a mock doesn't call actual methods unless explicitly told to do so or you spy instead of mock.

Reply:
Well now I'm baffled — I'm sure I hit a situation with a partial mock where I had non-mocked methods executing. I must be mis-remembering something ¯\_(ツ)_/¯